core/src/main/java/org/infinispan/topology/LocalTopologyManagerImpl.java

```diff
@@ -813,5 +852,22 @@ CacheTopology getStableTopology() {
    void setStableTopology(CacheTopology stableTopology) {
       this.stableTopology = stableTopology;
       partitionHandlingManager.onTopologyUpdate(currentTopology);
+      if (stableTopology != null) stable.complete(null);
+   }
+
+   List<Address> knownMembers() {
+      return Immutables.immutableListConvert(knownMembers);
```

Comment:
This seems extraneous, we should just store the immutable List in the field. Also I am not sold do we really need the Set to remove duplicates or can we trust that the invokers of `setCurrentMembers` will send us a list without duplicates? I would rather not have this extra Set used here if possible.